src/quequations.jl

```diff
@@ -1,61 +1,158 @@
+
+@doc """
+Lindblad operator construct from the `Hamiltonian` and `collapse operators`
+
+Input Parameters :
+
+`hamiltonian` : Hamiltonain of the system
+`collapse_ops` : Collapse operators
+""" ->
+function lindblad_op(hamiltonian::QuBase.AbstractQuMatrix, collapse_ops::Vector)
+    nb = size(coeffs(hamiltonian), 1)
+    nlop = length(collapse_ops)
+    heff = zeros(typeof(im*one(eltype(hamiltonian))), size(coeffs(hamiltonian)))
+    for l=1:nlop
+        heff = heff + 0.5*coeffs(collapse_ops[l])'*coeffs(collapse_ops[l])
+    end
     SI = Array(Int,0)
     SJ = Array(Int,0)
-    Lvals = Array(typeof(im*one(eltype(h))),0)
+    Lvals = Array(typeof(im*one(eltype(hamiltonian))),0)
     for m=1:nb
         for n=1:nb
             for i=1:nb
                 for j=1:nb
                     sm = (n-1)*nb + m
                     sj = (j-1)*nb + i
-                    lv = zero(Complex128)
+                    lv = zero(eltype(Lvals))
+                    for l=1:nlop
+                        lv = lv + collapse_ops[l][m,i]*collapse_ops[l]'[j,n]
+                    end
                     if j==n
-                        lv = lv - im*h[m,i]
+                        lv = lv - im*hamiltonian[m,i] - heff[m,i]
                     end
                     if i==m
-                        lv = lv + im*h[j,n]
+                        lv = lv + im*hamiltonian[j,n] - heff[j,n]
                     end
                     if real(lv)!=0 || imag(lv)!=0
                         push!(SI, sm)
                         push!(SJ, sj)
-                        push!(Lvals, lv)
+                        push!(Lvals, im*lv)
                     end
                 end
             end
         end
     end
-    return QuBase.similar_type(h)(sparse(SI, SJ, Lvals, nb*nb, nb*nb))
+    return QuBase.similar_type(hamiltonian)(sparse(SI, SJ, Lvals, nb*nb, nb*nb))
 end
 
+@doc """
+Liouvillian operator construct from the `Hamiltonian` and passing an empty array to `lindblad_op`
+
+Input Parameters :
+
+`hamiltonian` : Hamiltonain of the system
+""" ->
+liouvillian_op(hamiltonian::QuBase.AbstractQuMatrix) = lindblad_op(hamiltonian, [])
+
 function liouvillian_tensor(h::QuBase.AbstractQuMatrix)
     return -im*(tensor(eye(h), h) - tensor(h',eye(h)))
 end
 
+@doc """
+Input Parameters : QuLiouvillevonNeumannEq type
+
+Returns the `liouvillian` of the `QuLiouvillevonNeumannEq` type.
+""" ->
 function operator(qu_eq::QuLiouvillevonNeumannEq)
     return qu_eq.liouvillian
 end
 
+@doc """
+Input Parameters : QuSchrodingerEq type
+
+Returns the `hamiltonian` of the `QuSchrodingerEq` type.
+""" ->
 function operator(qu_eq::QuSchrodingerEq)
     return qu_eq.hamiltonian
 end
 
+@doc """
+Input Parameters : QuLindbladMasterEq type
+
+Returns the `lindblad operator` of the `QuLindbladMasterEq` type.
+""" ->
+function operator(qu_eq::QuLindbladMasterEq)
+    return lindblad_op(qu_eq.hamiltonian, qu_eq.collapse_ops)
```

Comment:
Sorry to ask this agin, but did we say the Linblad matrix should be cached and stored in the `QuLindbladmasterEq` to be consistent with the Liouville case?